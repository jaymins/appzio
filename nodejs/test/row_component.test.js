const { Row, Text } = require('../src/components')
const { Submit } = require('../src/actions')
const expect = require('expect')

describe('Row Component', () => {
  it('Creates a Row that matches the snapshot', () => {
    const row = Row()
    expect(row).toMatchSnapshot()
  })
  it('Creates a row containing the given components', () => {
    const text1 = Text('test text 1')
    const text2 = Text('test text 2')

    const row = Row([ text1, text2 ])

    expect(row.row_content.length).toBe(2)
  })
  it('Creates an empty array of contents if no components are passed', () => {
    const row = Row()
    expect(row.row_content.length).toBe(0)
  })
  it('Creates a row with params and inline styles', () => {
    const text1 = Text('test text 1')
    const text2 = Text('test text 2')

    const row = Row(
      [ text1, text2 ],
      { onclick: Submit('submit') },
      { backgroundColor: '#000000' }
    )

    expect(row.onclick).toBeDefined()
    expect(row.style_content['background-color']).toEqual('#000000')
  })
})
