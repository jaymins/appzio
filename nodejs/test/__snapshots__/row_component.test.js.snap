// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component Creates a Row that matches the snapshot 1`] = `
Object {
  "content": null,
  "row_content": Array [],
  "style_content": Object {},
  "type": "row",
}
`;
